core/src/utils/overlays.ts

```diff
@@ -327,34 +329,13 @@ export const getOverlays = (doc: Document, selector?: string): HTMLIonOverlayEle
     .filter(c => c.overlayIndex > 0);
 };
 
-/**
- * Gets the top-most/last opened
- * overlay that is currently presented.
- */
-const getTopOpenOverlay = (doc: Document): HTMLIonOverlayElement | undefined => {
-  const overlays = getOverlays(doc);
-  for (let i = overlays.length - 1; i >= 0; i--) {
-    const overlay = overlays[i];
-
-    /**
-     * Only consider overlays that
-     * are presented. Presented overlays
-     * will not have the .overlay-hidden
-     * class on the host.
-     */
-    if (!overlay.classList.contains('overlay-hidden')) {
-      return overlay;
-    }
-  }
-
-  return;
-}
-
 export const getOverlay = (doc: Document, overlayTag?: string, id?: string): HTMLIonOverlayElement | undefined => {
   const overlays = getOverlays(doc, overlayTag);
-  return (id === undefined)
-    ? overlays[overlays.length - 1]
-    : overlays.find(o => o.id === id);
+  if (id === undefined) {
+    const visibleOverlays = overlays.filter(o => !isOverlayHidden(o));
+    return visibleOverlays[visibleOverlays.length - 1];
+  }
+  return overlays.find(o => o.id === id);
```

Comment:
This behavior seems confusing to me. If there is no ID, then this will exclude hidden overlays. However, if there is an ID, then this will include hidden overlays. Having the include/exclude behavior determined by the ID does not seem right.

Perhaps there should be a parameter that lets you indicate whether or not you want to include all overlays regardless of hidden state? While I do consider this to be a bug fix as the original behavior seems wrong, the new behavior could be considered breaking by some, so adding this may be a good compromise.

Reply:
Added a parameter to either include or exclude hidden overlays. Default behavior is to include hidden overlays to be a non-breaking change. Let me know if that aligns with how you were thinking. 

Would love to consider moving to an object-based parameter for configuring the `getOverlay` method instead of individual parameters for each option in v7.